### PR TITLE
Dispose of MemoryPool2; suppress its finalizers

### PR DESCRIPTION
**Pull request by benaadams**
This is to unpin the MemoryPoolSlab2 buffers and SuppressFinalize on all the MemoryPoolBlock2 objects

### CODE REVIEW

**Comment by benaadams** [comment]
Was rebased on #337 to reduce probability of CI failures

File: test/Microsoft.AspNet.Server.KestrelTests/TestInput.cs [hunk 1]
```diff
@@ -9,21 +9,28 @@
 
 namespace Microsoft.AspNet.Server.KestrelTests
 {
-    class TestInput : IConnectionControl, IFrameControl
```
Comment (halter73) [inline test/Microsoft.AspNet.Server.KestrelTests/TestInput.cs]: Doesn't this require change MessageBodyTests?

Comment (benaadams) [inline test/Microsoft.AspNet.Server.KestrelTests/TestInput.cs]: Added

**Comment by benaadams** [comment]
This is probably better #402

**Pull request closed without merging**